django/http/request.py

```diff
@@ -249,28 +260,28 @@ def _load_post_and_files(self):
             self._mark_post_parse_error()
             return
 
-        if self.META.get('CONTENT_TYPE', '').startswith('multipart/form-data'):
-            if hasattr(self, '_body'):
-                # Use already read data
-                data = BytesIO(self._body)
-            else:
-                data = self
-            try:
+        try:
+            if self.META.get('CONTENT_TYPE', '').startswith('multipart/form-data'):
+                if hasattr(self, '_body'):
+                    # Use already read data
+                    data = BytesIO(self._body)
+                else:
+                    data = self
                 self._post, self._files = self.parse_file_upload(self.META, data)
-            except MultiPartParserError:
-                # An error occurred while parsing POST data. Since when
-                # formatting the error the request handler might access
-                # self.POST, set self._post and self._file to prevent
-                # attempts to parse POST data again.
-                # Mark that an error occurred. This allows self.__repr__ to
-                # be explicit about it instead of simply representing an
-                # empty POST
-                self._mark_post_parse_error()
-                raise
-        elif self.META.get('CONTENT_TYPE', '').startswith('application/x-www-form-urlencoded'):
-            self._post, self._files = QueryDict(self.body, encoding=self._encoding), MultiValueDict()
-        else:
-            self._post, self._files = QueryDict('', encoding=self._encoding), MultiValueDict()
+            elif self.META.get('CONTENT_TYPE', '').startswith('application/x-www-form-urlencoded'):
+                self._post, self._files = QueryDict(self.body, encoding=self._encoding), MultiValueDict()
+            else:
+                self._post, self._files = QueryDict(encoding=self._encoding), MultiValueDict()
+        except (MultiPartParserError, SuspiciousOperation):
```

Comment:
Ohh silly me. Yes, only exceptions. My though was that it might make sense to mark `_mark_post_parse_error()` regardless of which exception was raised. My thinking is if any error happens here, it would happen again in the debug traceback.